includes/discovery/loadbalancers/f5-gtm.inc.php

```diff
@@ -0,0 +1,187 @@
+<?php
+/*
+ * LibreNMS module to display F5 GTM Wide IP Details
+ *
+ * Adapted from F5 LTM module by Darren Napper
+ * Copyright (c) 2016 Aaron Daniels <aaron@daniels.id.au>
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or (at your
+ * option) any later version.  Please see LICENSE.txt at the top level of
+ * the source code distribution for details.
+ */
+
+// Define some error messages
+use LibreNMS\Util\IP;
+
+$error_poolaction = array();
+$error_poolaction[0] = "Unused";
+$error_poolaction[1] = "Reboot";
+$error_poolaction[2] = "Restart";
+$error_poolaction[3] = "Failover";
+$error_poolaction[4] = "Failover and Restart";
+$error_poolaction[5] = "Go Active";
+$error_poolaction[6] = "None";
+
+$component = new LibreNMS\Component();
+$components = $component->getComponents($device['device_id']);
+
+// We only care about our device id.
+$components = $components[$device['device_id']];
+
+// We extracted all the components for this device, now lets only get the LTM ones.
+$keep = array();
+$types = array($module, 'bigip', 'f5-gtm-wide', 'f5-gtm-pool');
+foreach ($components as $k => $v) {
+    foreach ($types as $type) {
+        if ($v['type'] == $type) {
+            $keep[$k] = $v;
+        }
+    }
+}
+$components = $keep;
+
+// Begin our master array, all other values will be processed into this array.
+$tblBigIP = array();
+
+//Check if GTM modeul is running and run object discovery.
+if ((snmp_get($device, 'sysModuleAllocationProvisionLevel.3.103.116.109', '-Ovqs', 'F5-BIGIP-SYSTEM-MIB')) !=none) {
+    $gtmWideIPEntry = snmpwalk_array_num($device, '1.3.6.1.4.1.3375.2.3.12.1.2.1', 0);
+    $gtmWideStatusEntry = snmpwalk_array_num($device, '1.3.6.1.4.1.3375.2.3.12.3.2.1', 0);
+    $gtmPoolEntry = snmpwalk_array_num($device, '1.3.6.1.4.1.3375.2.3.6.2.3.1.1', 0);
+}
+
+/*
+ * False == no object found - this is not an error, OID doesn't exist.
+ * null  == timeout or something else that caused an error, OID may exist but we couldn't get it.
+ */
+
+if (!is_null($gtmWideIPEntry) || !is_null($gtmWideStatusEntry) || !is_null($gtmPoolEntry)) {
```

Comment:
Don't these need to be && if you want them all to match?

Reply:
You don't need to have a pool attached to a wide IP for it to function so there could be situations where people are using the device to service static DNS records as well as dynamic ones based on pool conditions.